test/e2e/broker_trigger_test.go

```diff
@@ -268,3 +269,141 @@ func TestBrokerWithConfig(t *testing.T) {
+
+		nonMatchingEventId := uuid.New().String()
+		eventId := uuid.New().String()
+
+		br := client.CreateBrokerOrFail(
+			brokerName,
+			resources.WithBrokerClassForBroker(kafka.BrokerClass),
+			WithExternalTopicForBroker(kafkaTopicName),
+		)
+
+		eventTracker, _ := recordevents.StartEventRecordOrFail(ctx, client, subscriber)
+
+		client.CreateTriggerOrFail(
+			triggerName,
+			resources.WithBroker(brokerName),
+			resources.WithSubscriberServiceRefForTrigger(subscriber),
+			func(trigger *eventing.Trigger) {
+				trigger.Spec.Filter = &eventing.TriggerFilter{
+					Attributes: map[string]string{
+						"source":   eventSource,
+						extension1: valueExtension1,
+						"type":     "",
+					},
+				}
+			},
+		)
+
+		client.WaitForAllTestResourcesReadyOrFail(ctx)
+
+		t.Logf("Sending events to %s/%s", client.Namespace, brokerName)
+
+		nonMatchingEvent := cloudevents.NewEvent()
+		nonMatchingEvent.SetID(eventId)
+		nonMatchingEvent.SetType(eventType)
+		nonMatchingEvent.SetSource(eventSource)
+		nonMatchingEvent.SetExtension(extension1, valueExtension1+"a")
+		if err := nonMatchingEvent.SetData(cloudevents.ApplicationJSON, []byte(eventBody)); err != nil {
+			t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+		}
+
+		client.SendEventToAddressable(
+			ctx,
+			senderName+"non-matching",
+			brokerName,
+			testlib.BrokerTypeMeta,
+			nonMatchingEvent,
+		)
+
+		eventToSend := cloudevents.NewEvent()
+		eventToSend.SetID(eventId)
+		eventToSend.SetType(eventType)
+		eventToSend.SetSource(eventSource)
+		eventToSend.SetExtension(extension1, valueExtension1)
+		if err := eventToSend.SetData(cloudevents.ApplicationJSON, []byte(eventBody)); err != nil {
+			t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+		}
+
+		client.SendEventToAddressable(
+			ctx,
+			senderName+"matching",
+			brokerName,
+			testlib.BrokerTypeMeta,
+			eventToSend,
+		)
+
+		eventTracker.AssertAtLeast(1, recordevents.MatchEvent(
+			HasId(eventId),
+			HasSource(eventSource),
+			HasType(eventType),
+			HasData([]byte(eventBody)),
+		))
+
+		eventTracker.AssertNot(recordevents.MatchEvent(
+			HasId(nonMatchingEventId),
+		))
+
+		config := &kafkatest.Config{
+			BootstrapServers:  testingpkg.BootstrapServersPlaintext,
+			ReplicationFactor: defaultReplicationFactor,
+			NumPartitions:     defaultNumPartitions,
+			Topic:             kafka.BrokerTopic(broker.TopicPrefix, br),
+		}
+
+		err := kafkatest.VerifyNumPartitionAndReplicationFactor(
+			client.Kube,
+			client.Tracker,
+			client.Namespace,
+			verifierName,
+			config,
+		)
+		if err != nil {
+			t.Errorf("failed to verify num partitions and replication factors: %v", err)
+		}
+	})
+}
+
+// WithExternalTopicForBroker returns a function that adds an
+// annotation to the given v1 (Kafka) Broker for using an external managed topic
+func WithExternalTopicForBroker(annotationValue string) BrokerOption {
```

Comment:
I think I will locate this somewhere else - but there is no real of a `lib`

That said, I think generally handy for upstream eventing is something like:

```
func WithCustomAnnotationForBroker(annotationKey, annotationValue string) BrokerOption {...}
```

Which _our_ `func` here could make use of ...